trymito.io/components/Glossary/PageTOC/PageTOC.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from 'react';
+import { classNames } from '../../../utils/classNames';
+import pageTOCStyles from './PageTOC.module.css';
+
+type Heading = {
+    level: string;
+    id: string;
+}
+
+const PageTOC = () => {
```

Comment:
This is a really interesting approach to this component. I don't have strong feelings, but I think it's legit. 

The only question is if the on scroll listener is way laggier than it needs to be... there's a rerender of the entire page for every scroll, which seems super unnecessary. 

What if instead, we debounced it? Even checking every 25ms or something is probably good.